doc/src/manual/calling-c-and-fortran-code.md

```diff
@@ -804,13 +778,9 @@ end
 ```
 
 Here, the input `p` is declared to be of type `Ref{gsl_permutation}`, meaning that the memory
-that `p` points to may be managed by Julia or by C. A pointer to memory allocated by C should
+that `p` points to may be managed by Julia or by C. A pointer to memory allocated by C should generally
 be of type `Ptr{gsl_permutation}`, but it is convertible using [`Base.cconvert`](@ref) and therefore
```

Comment:
```suggestion
be of type `Ptr{gsl_permutation}`. But `Ptr` is a subtype of `Ref`, and therefore `Ref` be used in any covariant (argument) context.
The `Ref` supertype will accept a `Ptr`, and treats it the same as if it had been restricted to just accept `Ptr`.
Additionally, this makes it possible to pass a handle to memory that has been allocated by Julia. 
```

Reply:
no, you can never use `pointer`. the rest sounds ok

Reply:
Okay, would I just pass `a` in that second example then?